File: lib/app/domain/sync_worker.dart
import 'sync_queue.dart';
import 'sync_trigger.dart';

class SyncWorker {
  SyncWorker({required SyncQueue queue}) : _queue = queue;

  final SyncQueue _queue;

  Future<int> onNetworkRecovered() {
    return _queue.syncPending(trigger: SyncTrigger.networkRecovered);
  }

  Future<int> onAppLaunch() {
    return _queue.syncPending(trigger: SyncTrigger.appLaunch);
  }

  Future<int> onPeriodic() {
    return _queue.syncPending(trigger: SyncTrigger.periodic);
  }
}

File: src/app/sync_worker.js
class SyncWorker {
  constructor({ queue }) {
    if (!queue || typeof queue.syncPending !== 'function') {
      throw new Error('queue.syncPending is required');
    }

    this.queue = queue;
  }

  async onNetworkRecovered() {
    return this.queue.syncPending({ trigger: 'network_recovered' });
  }

  async onAppLaunch() {
    return this.queue.syncPending({ trigger: 'app_launch' });
  }

  async onPeriodic() {
    return this.queue.syncPending({ trigger: 'periodic' });
  }
}

module.exports = {
  SyncWorker
};
